### PR TITLE
Add sentry session tracking

### PR DESCRIPTION
**Pull request by raubitsj**
Description
-----------
Problem:
We have information about failures seen by wandb library users but we don't have a good way of knowing the percentage of users that encounter a given problem.   The goal is to calculate a problem rate metric:

ProblemRate = NumberOfUsersExperiencingAProblem / TotalNumberOfUsers

The service sentry.io collects problems, and number of users.  but we have a blind spot within sentry about total number of users.   We have other data sources which we could use to get this denominator but then the way data is being captured could be inconsistent.   The larger project to work on this is/will be:  [Internal Notion Doc](https://www.notion.so/wandbai/Problem-rate-local-telemetry-0997b4ac34f642a29835c66daaf87e8b)

Sentry has two endpoints we would like to use:
- [store](https://develop.sentry.dev/sdk/store/)
- [envelope](https://develop.sentry.dev/sdk/envelopes/)

Currently we are only using the store endpoint.
Envelope can be used to send session informaiton.

Caveats:
- The default thread which saves session information runs with a sleep timer of 60 seconds and might prevent runs shorter than 60 seconds from being captured
- We might need to be careful how we define a session, is the internal process and user process a single session, is it ok if they are two sessions?   The ProblemRate per user should be not impacted by that distinction
- Should we use the sentry auto session interface?

Partially answered questions:
- How was this working before? Before June 2022 we were using the global sentry hub so if our user was also using sentry they might have configured it to use a session which would explain why we had data for a while.

Possible future changes:
- We might change our sentry sdk project so we have cleaner data once this change goes out and is good (we already have sdk-old)
- We might want to add our own exception endpoint so we can collect data more inline.  We will use sentry for digging into details but use our own internal metric to track improvements over time.   We should validate that they give similar answers or understand why they dont.
- If we are adding our own exception endpoint, we might add a telemetry endpoint to keep track of other situations like how often certain features are used in the command line.

TODO:
- [x] Debug some more
- [x] Add envelope support to mockserver / yea

Requires:
- https://github.com/wandb/yea-wandb/pull/96

Testing
-------
Will be tested with yea

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

**Review by dmitryduev — APPROVED** [review]
this easy?? sick!!

File: wandb/util.py [hunk 1]
```diff
@@ -256,6 +256,10 @@ def get(key: str) -> Any:
             if value is not None and value != "":
                 scope.set_tag(tag, value)
 
+    # Track session so we can get metrics about error free rate
+    if sentry_hub:
+        sentry_hub.start_session()
```
Comment (dmitryduev) [inline wandb/util.py:261]: Could this be tested easily or do you need to dig deep into the sentry internals for that?
Also, there's no need to do "end_session" or something? :)

Comment (raubitsj) [reply wandb/util.py:262]: We end up starting two sessions for each user process and the internal/service process.  This should be ok as we are mainly interested in crash free users.  The session is run by a timer, the default update rate is every 60 seconds so we are going to miss short sessions.   Let's see how this release goes and see if we need to do more to cover some of these caveats.

**Review by dmitryduev — APPROVED** [review]
LGTM

**Pull request merged**